File: substitution.py
def get_substitution(text: str, sub_map: dict[str, str]) -> str:
    substituted: str = ""
    for char in text:
        substitution: str | None = sub_map.get(char)

        if substitution is None:
            substitution = "<INV_CHAR>"

        substituted += substitution

    return substituted


def get_substitution_over(text: str, sub_maps: list[dict[str, str]]) -> list[str]:
    subs: list[str] = [get_substitution(text, sub_map) for sub_map in sub_maps]

    return subs

File: crypt/app/util/substitution.ts
export function getSubstitution(text: string, subMap: Map<string, string>): string {
	let substituted = "";

	for (let i = 0; i < text.length; i++) {
		let substitution: string | undefined = subMap.get(text[i]);

		if (!substitution) {
			substitution = "<INV_CHAR>";
		}

		substituted += substitution;
	}

	return substituted;
}

export function getSubstitutionOver(text: string, subMaps: Array<Map<string, string>>): Array<string> {
	const subs: Array<string> = [];

	subMaps.forEach((subMap) => {
		const sub = getSubstitution(text, subMap);
		subs.push(sub);
	})

	return subs;
}
